app/frontend/src/common/authenticate.js

```diff
@@ -0,0 +1,76 @@
+/**
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+import { SET_KEYCLOAK } from '@/registry/store/mutations.types.js'
+import { store } from '@/registry/store'
+import Keycloak from 'keycloak-js'
+import ApiService from '@/common/services/ApiService.js'
+import Vue from 'vue'
+
+export default {
+  getInstance: function () {
+    if (!Vue.prototype.$keycloak) {
+      // TODO: This has to read settings from environment configuration!
+      Vue.prototype.$keycloak = Keycloak({
+        'realm': 'gwells',
+        'url': 'https://dev-sso.pathfinder.gov.bc.ca/auth',
+        'clientId': 'webapp-dev-local'
+      })
+    }
+    return Vue.prototype.$keycloak
+  },
+
+  authenticate: function () {
+    return new Promise((resolve, reject) => {
+      const instance = this.getInstance()
+      if (instance.authenticated && ApiService.hasAuthHeader() && !instance.isTokenExpired(0)) {
+        resolve() // We've already authenticated, have a header, and we've not expired.
+      } else {
+        // Attempt to retrieve a stored token, this may avoid us having to refresh the page.
+        const token = localStorage.getItem('token')
+        const refreshToken = localStorage.getItem('refreshToken')
+        const idToken = localStorage.getItem('idToken')
+        instance.init({
+          onLoad: 'check-sso',
+          checkLoginIframe: true,
+          timeSkew: 10, // Allow for some deviation
+          token,
+          refreshToken,
+          idToken }
+        ).success((result) => {
+          store.commit(SET_KEYCLOAK, instance)
+          if (instance.authenticated) {
+            // We may have been authenticated, but the token could be expired.
+            if (instance.isTokenExpired(0)) {
+              // The token is expired!
+              result = false // As far as we're concerned, this means authentication failed.
```

Comment:
possibly reject promise instead?

Reply:
I'm leaving as is right now - mirroring keycloak having true/false for the init - but it's something to relook. We could probably just get rid of this promise entirely.